model/configuration_deployment.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package model
+
+import (
+	"time"
+
+	"github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/pkg/errors"
+)
+
+// ConfigurationDeploymentConstructor represent input data needed for creating new Configuraiont Deployment
+type ConfigurationDeploymentConstructor struct {
+	// Deployment name, required
+	Name string `json:"name"`
+
+	// A string containing a configuration object.
+	// The deployments service will use it to generate configuration
+	// artifact for the device.
+	// The artifact will be generated when the device will ask
+	// for an update.
+	Configuration string `json:"configuration"`
```

Comment:
I'm pretty sure you had the Retries here in the enterprise version. Can we add it here, as well?
We can simply ignore the field in the OS version, but this simplifies my life in deviceconfig/workflows. :)

Reply:
sure; would it be enough to have it here or should we have it in the deployment object too?

Reply:
done

Reply:
Thinking about it twice, as long as we do not return 400 bad request because we set the "retries" attribute, I'm fine if we do not add the field. I believe in this case it will be ignored when unmarshalling the request, right?

Reply:
yes, it will, but maybe it's safer to have it there in case we'll treat additional fields as client errors at some point